app/src/androidTest/java/com/github/se/project/ui/lesson/TutorMatchingTest.kt

```diff
@@ -0,0 +1,253 @@
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.spy
+import org.mockito.Mockito.verify
+import org.mockito.kotlin.any
+import org.mockito.kotlin.whenever
+
+class TutorMatchingScreenTest {
+
+  @get:Rule val composeTestRule = createComposeRule()
+
+  private lateinit var profilesRepository: ProfilesRepository
+  private lateinit var listProfilesViewModel: ListProfilesViewModel
+
+  private lateinit var lessonRepository: LessonRepository
+  private lateinit var lessonViewModel: LessonViewModel
+
+  private lateinit var navigationActions: NavigationActions
+  private lateinit var navController: NavHostController
+
+  private val profileFlow =
+      MutableStateFlow(
+          Profile(
+              uid = "uid",
+              googleUid = "googleUid",
+              firstName = "First",
+              lastName = "Last",
+              phoneNumber = "1234567890",
+              role = Role.STUDENT,
+              section = Section.GM,
+              academicLevel = AcademicLevel.MA2,
+              languages = listOf(),
+              subjects = listOf(),
+              schedule = listOf()))
+
+  private val lessonFlow =
+      MutableStateFlow(
+          Lesson(
+              id = "lessonId",
+              title = "Math Lesson",
+              description = "Algebra",
+              subject = Subject.ANALYSIS,
+              languages = listOf(Language.ENGLISH),
+              tutorUid = listOf("tutor123"),
+              studentUid = "student123",
+              minPrice = 10.0,
+              maxPrice = 50.0,
+              timeSlot = "2024-12-12T12:00",
+              status = LessonStatus.MATCHING,
+              latitude = 0.0,
+              longitude = 0.0))
+
+  private val tutorsFlow =
+      MutableStateFlow(
+          listOf(
+              Profile(
+                  uid = "tutor123",
+                  googleUid = "googleUid",
+                  firstName = "Tutor",
+                  lastName = "One",
+                  phoneNumber = "0987654321",
+                  role = Role.TUTOR,
+                  section = Section.IN,
+                  academicLevel = AcademicLevel.MA1,
+                  languages = listOf(Language.ENGLISH),
+                  subjects = listOf(Subject.ANALYSIS),
+                  schedule = listOf(),
+                  price = 30)))
+
+  @Before
+  fun setup() {
+    // Mock dependencies
+    profilesRepository = mock(ProfilesRepository::class.java)
+    lessonRepository = mock(LessonRepository::class.java)
+    navController = mock()
+
+    listProfilesViewModel = ListProfilesViewModel(profilesRepository)
+    listProfilesViewModel = spy(listProfilesViewModel)
+
+    lessonViewModel = LessonViewModel(lessonRepository)
+    lessonViewModel = spy(lessonViewModel)
+
+    navigationActions = NavigationActions(navController)
+    navigationActions = spy(navigationActions)
+
+    // Stub navigation actions
+    doNothing().`when`(navigationActions).navigateTo(anyString())
```

Comment:
we could test difference response from the mocked Navhost but for the moment it is enough to test our limited feature that we implemented